### PR TITLE
improve typings on "getting started"

### PR DESCRIPTION
**Pull request by KATT**

### CODE REVIEW

File: pages/getting_started/index.tsx [hunk 1]
```diff
@@ -609,109 +613,116 @@ export default function Onboarding(props: OnboardingProps) {
...
+      avatar: true,
+      timeZone: true,
+      completedOnboarding: true,
+    },
+  });
+  if (!user) {
+    throw new Error(`Signed in as ${session.user.id} but cannot be found in db`);
+  }
 
-  if (session) {
```
Comment (KATT) [inline pages/getting_started/index.tsx:618]: removed this massive block and replaced it with 

```ts
  if (!session?.user?.id) {
    return {
      redirect: {
        permanent: false,
        destination: "/auth/login",
      } as const,
    };
  }
```

File: pages/getting_started/index.tsx [hunk 2]
```diff
@@ -29,6 +30,7 @@ import { Integration } from "pages/integrations";
 import { AddCalDavIntegrationRequest } from "../../lib/integrations/CalDav/components/AddCalDavIntegration";
 import classnames from "classnames";
 import { ArrowRightIcon } from "@heroicons/react/outline";
+import { getSession } from "@lib/auth";
```
Comment (KATT) [inline pages/getting_started/index.tsx:33]: using our own `getSession()` that has some more meta data

File: pages/getting_started/index.tsx [hunk 3]
```diff
@@ -555,6 +557,8 @@ export default function Onboarding(props: OnboardingProps) {
                 Step {currentStep + 1} of {steps.length}
               </Text>
 
+              {error && <ErrorAlert {...error} />}
```
Comment (KATT) [inline pages/getting_started/index.tsx:560]: this fixes the lint issue

**Review by KATT — COMMENTED** [review]
See comments - highlighted the changes

**Review by pumfleet — APPROVED** [review]

File: pages/getting_started/index.tsx (hunk 1 above)
Comment (femyeda) [reply pages/getting_started/index.tsx:621]: Smart fix on this.

**Review by femyeda — APPROVED** [review]
We should add a type and a constant for this very common integrations array as well.

**Pull request merged**